scalyr_agent/builtin_monitors/windows_event_log_monitor.py

```diff
@@ -625,18 +636,92 @@ def log_event(self, event):
 
         self._logger.emit_value("EventLog", provider, extra_fields=vals)
 
-        self.__bookmark_lock.acquire()
+        self._bookmark_lock.acquire()
         try:
             if "Channel" in vals:
                 channel = vals["Channel"]
-                bookmark = None
-                if channel not in self.__bookmarks:
-                    self.__bookmarks[channel] = win32evtlog.EvtCreateBookmark(None)
+                if channel not in self._bookmarks:
+                    self._bookmarks[channel] = win32evtlog.EvtCreateBookmark(None)
+                win32evtlog.EvtUpdateBookmark(self._bookmarks[channel], event)
+        finally:
+            self._bookmark_lock.release()
+
+
+class NewJsonApi(NewApi):
+    def __init__(self, config, logger, channels):
+        super(NewJsonApi, self).__init__(config, logger, channels)
 
-                bookmark = self.__bookmarks[channel]
-                win32evtlog.EvtUpdateBookmark(bookmark, event)
+        self._render_context = win32evtlog.EvtCreateRenderContext(
+            win32evtlog.EvtRenderContextSystem
+        )
+
+    def log_event(self, event):
+        values = win32evtlog.EvtRender(
+            event, win32evtlog.EvtRenderEventValues, Context=self._render_context
+        )
+
+        metadata = win32evtlog.EvtOpenPublisherMetadata(
+            values[win32evtlog.EvtSystemProviderName][0]
+        )
+
+        event_json = xmltodict.parse(
+            win32evtlog.EvtFormatMessage(
+                metadata, event, win32evtlog.EvtFormatMessageXml
```

Comment:
@dariocalia mentioned JSON format may also be available by the win32 API, but that doesn't appear to be the case - https://docs.microsoft.com/en-us/windows/win32/api/winevt/ne-winevt-evt_format_message_flags

Reply:
Yep, I looked pretty hard & didn't find an approach to get json directly from the win32 api.

It's reassuring to know that Elastic does an xml-to-json conversion as well: https://github.com/elastic/beats/blob/main/winlogbeat/eventlog/wineventlog.go#L323